lms/djangoapps/django_comment_client/tests/test_utils.py

```diff
@@ -170,6 +173,28 @@ def assertThreadCorrect(thread, discussion, expected_title):  # pylint: disable=
         assertThreadCorrect(threads[0], self.discussion1, "Chapter / Discussion 1")
         assertThreadCorrect(threads[1], self.discussion2, "Subsection / Discussion 2")
 
+    @ddt.data(ModuleStoreEnum.Type.mongo, ModuleStoreEnum.Type.split)
+    def test_get_accessible_discussion_modules(self, modulestore_type):
+        """
+        Tests that the accessible discussion modules having no parents do not get fetched for split modulestore.
+        """
+        # Create a course using modulestore passed as an argument.
+        course = CourseFactory.create(default_store=modulestore_type)
+
+        # Add an orphan discussion module to that course
+        orphan = course.id.make_usage_key('discussion', 'orphan_discussion')
+        self.store.create_item(self.user.id, orphan.course_key, orphan.block_type, block_id=orphan.block_id)
+
+        # Assert that the discussion module is an orphan.
+        self.assertIn(orphan, self.store.get_orphans(course.id))
+
+        if modulestore_type == ModuleStoreEnum.Type.split:
```

Comment:
Instead of doing an if/else here, add `expected_discussion_modules` to ddt, so you can just do:
```python
self.assertEqual(len(utils.get_accessible_discussion_modules(course, self.user)), expected_discussion_modules)
```
Can you also add one discussion module that's _in_ the course tree, so that we can make sure that `utils.get_accessible_discussion_modules` does return discussion modules in the course tree